docs/contracts-intro.md

```diff
@@ -2,11 +2,159 @@
 
 ---8<-- 'includes/data-builder-danger-header.md'
 
-This page will introduce OpenSAFELY Contracts, explaining:
+## OpenSAFELY Contracts are a clinical data specification
 
-* what they are
-* the process by which new contracts could be proposed
-* where [details of existing contracts will be
-  found](contracts-reference.md)
+*OpenSAFELY Contracts* are an important OpenSAFELY concept that is
+introduced with Data Builder.
+
+This page describes:
+
+* what OpenSAFELY Contracts are
+* how OpenSAFELY Contracts relate to dataset definitions
+
+Each OpenSAFELY Contract is a formal specification — [described in
+code](https://github.com/opensafely-core/databuilder/blob/main/databuilder/contracts/)
+— relating to a particular clinical data domain. OpenSAFELY Contracts
+are authored by the OpenSAFELY platform developers: the Contract designs
+are informed by discussion and feedback from [data
+providers](data-provider-integration.md) and researchers.
+
+Researchers write dataset definitions to access data tables from
+OpenSAFELY backends. All of the data tables available from OpenSAFELY
+satisfy a single OpenSAFELY Contract, regardless of which data provider
+operates the backend.
+
+## The OpenSAFELY Contracts reference
+
+The [OpenSAFELY Contracts reference](contracts-reference.md) has full
+details of each individual Contract. When writing a dataset definition,
+researchers should consult [the OpenSAFELY Contracts
+reference](contracts-reference.md) to see:
+
+* what data is available in OpenSAFELY backends
+* how the data is structured
+* what that data represents
+
+!!! warning
+    OpenSAFELY Contracts apply only when accessing OpenSAFELY data using
```

Comment:
How about something like "OpenSAFELY contracts are a concept introduced with Data Builder, the replacement for our legacy cohort-extractor system"